detox/src/utils/environment.js

```diff
@@ -157,12 +158,16 @@ function getDetoxVersion() {
   return require(path.join(__dirname, '../../package.json')).version;
 }
 
-async function getFrameworkPath() {
-  const detoxVersion = this.getDetoxVersion();
+async function setFrameworkPath() {
+  const detoxVersion = getDetoxVersion();
   const sha1 = (await exec(`(echo "${detoxVersion}" && xcodebuild -version) | shasum | awk '{print $1}'`)).stdout.trim();
   return `${DETOX_LIBRARY_ROOT_PATH}/ios/${sha1}/Detox.framework`;
 }
 
+function getFrameworkPath() {
+  return IOS_FRAMEWORK_PATH;
```

Comment:
I'd insist on lazy calling setFrameworkPath here, if the promise is not initialized.
Modules with side effects upon require are less pleasant to test. 

Also this code might break on Windows and Linux if you call it for no good reason.